### DISCUSSION TITLE
Binary files corresponds to a CPU architecture error

### DISCUSSION THREAD

**Discussion by threo9** [coder/code-server#3483] [Q&A] [answered]
I'm running code-server(It's the latest version) on my x86 Linux server. 
I have installed the extension named "C/C++ Compile Run".
(Both from the vsc official market by vsix file and from your market I've try)
But the extension always download the file for ARM64 for me,
![image](https://user-images.githubusercontent.com/69025582/119606051-6e167e00-be24-11eb-8425-059411a177da.png)

It cannot run on my x86 server, so the extension showed me the language server has crashed.
![image](https://user-images.githubusercontent.com/69025582/119606162-a027e000-be24-11eb-88c2-c687e272697b.png)

I don't know how to make it works without replacing the wrong file, I want it can download the correct file.
So, could you please tell me what should I do? Need I submit an issue to the extension program?
(I'm a Chinese student, so maybe my English is too poor to express it clearly,I'm very sorry about it.
You can ask it to me at any time if there is any unclear statement, Thx!)
I've tried on another x86 server, it's also cannot run.

**Comment by code-asher** [comment]
Ah yup our marketplace has an issue right now: https://github.com/cdr/code-server/issues/2120

I think the easiest workaround is to go to the releases page on GitHub and download `cpptools-linux.vsix`: https://github.com/microsoft/vscode-cpptools/releases

**Reply by threo9** [reply]
Got it,thank you.

**Comment by ghost** [comment, marked as answer]
Im using a Docker container to run code-server `3.10.2` using [cdr/deploy-code-server](https://github.com/cdr/deploy-code-server/) on https://railway.app. I ran into the same problem but with the C/C++ Support extension from Microsoft. Instead of manually downloading and installing everytime, I added this to my Dockerfile:
```Dockerfile
RUN sudo apt-get install --no-install-recommends -y wget

RUN sudo wget https://github.com/microsoft/vscode-cpptools/releases/download/1.4.1/cpptools-linux.vsix -P /home/coder/vsix/

RUN code-server --install-extension /home/coder/vsix/cpptools-linux.vsix
```
